sdk/extensions/Azure.Extensions.AspNetCore.DataProtection.Blobs/src/AzureBlobXmlRepository.cs

```diff
@@ -71,32 +71,30 @@ public void StoreElement(XElement element, string friendlyName)
             Task.Run(() => StoreElementAsync(element)).GetAwaiter().GetResult();
         }
 
-        private static XDocument CreateDocumentFromBlob(byte[] blob)
+        private static XDocument CreateDocumentFromBlobData(BlobData blobData)
         {
-            using (var memoryStream = new MemoryStream(blob))
+            if (blobData == null || blobData.BlobContents.Length == 0)
             {
-                var xmlReaderSettings = new XmlReaderSettings()
-                {
-                    DtdProcessing = DtdProcessing.Prohibit, IgnoreProcessingInstructions = true
-                };
+                return new XDocument(new XElement(RepositoryElementName));
+            }
 
-                using (var xmlReader = XmlReader.Create(memoryStream, xmlReaderSettings))
-                {
-                    return XDocument.Load(xmlReader);
-                }
+            using var memoryStream = new MemoryStream(blobData.BlobContents);
+
+            var xmlReaderSettings = new XmlReaderSettings()
+            {
+                DtdProcessing = DtdProcessing.Prohibit, IgnoreProcessingInstructions = true
```

Comment:
Nit: put property assignments on separate lines, and add terminating comma to last line (makes future diffs cleaner).